src/org/zaproxy/zap/extension/websocket/WebSocketProxy.java

```diff
@@ -1001,4 +1012,53 @@ public boolean isAllowAPI() {
 	public void setAllowAPI(boolean allowAPI) {
 		this.allowAPI = allowAPI;
 	}
+	
+	/**
+	 * Return an instance of {@link HandshakeConfig} which is used to establish a new WebSocket Connection
+	 * see {@link ServerConnectionEstablisher#send(HandshakeConfig)} and {@link WebSocketProxy#reEstablishConnection(HandshakeConfig)}
+	 * The {@link HandshakeConfig} includes the HttpHandshake of the current connection. In addition, the method adds the
+	 * {@link WebSocketObserver} and {@link WebSocketSenderListener}  of the current connection to HandshakeConfig
+	 *
+	 * @return the HandshakeConfig
+	 */
+	public HandshakeConfig getHandShakeConfig(boolean isFollowRedirection, boolean isStateRequest){
```

Comment:
The parameters could be removed, they can be tweaked by the caller in the returned config, if needed. (We need a new `HandshakeConfig` constructor with a `HttpMessage` though.)